src/Checker/OrderCustomerRelationChecker.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\CustomerReorderPlugin\Checker;
+
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+
+final class OrderCustomerRelationChecker implements OrderCustomerRelationCheckerInterface
+{
+    public function wasOrderPlacedByCustomer(OrderInterface $order, CustomerInterface $customer): bool
+    {
+        /** @var CustomerInterface $orderCustomer */
+        $orderCustomer = $order->getCustomer();
+
+        return
+            null != $customer &&
+            null != $orderCustomer &&
```

Comment:
Comparison should be strict `!==`

Reply:
You should discuss that with phpstan :D 

```
Strict comparison using !== between null and Sylius\Component\Core\Model\CustomerInterface will always evaluate to true.
```

$orderCustomer variable is typehinted a few lines above